### ISSUE TITLE
`--disable_perceptual_optimization` produces unexpectedly large files

### ISSUE THREAD

**Issue by cchang326** [libjxl/libjxl#3896]
**Describe the bug**
Encoding an image with and without the `--disable_perceptual_optimization` option produces wildly different file sizes. With the option, compressed file is 3.5x the size without the option.

**To Reproduce**
1. Encode an image with perceptual optimization disabled.
`\bin\jxl\cjxl.exe kodim05.png rgb_percep0.jxl -q 90 -e 3 --disable_perceptual_optimizations`

2. Encode the same image without the option:
`\bin\jxl\cjxl.exe kodim05.png rgb.jxl -q 90 -e 3`

3. Observe the compressed file sizes (in bytes):

140,794 rgb.jxl
515,229 rgb_percep0.jxl

**Expected behavior**
File size shouldn't be so drastically different.

**Screenshots**
If applicable, add screenshots or example input/output images to help explain your problem.

**Environment**
 - OS: [e.g. Windows]
 - Compiler version: [e.g. clang 11.0.1]
 - CPU type: [e.g. x86_64]
 - cjxl/djxl version string: [e.g. cjxl [v0.3.7 | SIMD supported: SSE4,Scalar]]

**Additional notes**
Example input image but I tried several images and they all behave the same.
![kodim05](https://github.com/user-attachments/assets/73fc2481-2acf-4224-a1b4-bfdfe46c70e3)

**Comment by jonnyawsom3** [comment]
That is expected. Without considering what the image looks like, the entire thing is encoded at the same highest quality level instead of putting the bits and bytes where they matter. It's mostly intended for non-visible images like satellite data, ect

**Comment by cchang326** [comment]
I discovered this when trying to disable color conversion to XYB when encoding gray scale images. Is there a way to disable XYB without disabling other perceptual optimizations?

**Comment by jonnyawsom3** [comment]
-C 0 should set it to RGB, but that may only work in combination with -m 1 to put it into Modular encoding, been a while since I used it

**Comment by cchang326** [comment]
I tried "-C 0 -m 1", its size is comparable when "--disable_perceptual_optimization". Is that expected?

120,703 gray.jxl
410,012 gray_C0.jxl
414,184 gray_percep0.jxl

**Comment by cchang326** [comment]
I forgot to mention I was using a gray scale image as input in my last message.

**Comment by cchang326** [comment]
<img width="384" alt="kodim05_gray" src="https://github.com/user-attachments/assets/cc678369-19b3-4a4c-8574-2b8302f0deed">

**Comment by jonnyawsom3** [comment]
Optimizing the PNG to 8bit greyscale means the output JXL is also 8bit greyscale. Likely related to #3415

**Comment by cchang326** [comment]
I'm not sure I understand your comment. Can you elaborate?

**Comment by Galaxy4594** [comment]
[Oxipng](https://github.com/shssoichiro/oxipng) and [ECT](https://github.com/fhanau/Efficient-Compression-Tool) are PNG optimizers that losslessly recompress PNGs. In this case, the colorspace of the b/w image is RGB, so these programs will detect that and convert it to Grayscale, saving a little bit of space. You can then feed the Grayscale PNG into cjxl and get a Grayscale JXL. 

Ideally, cjxl should be able to detect when an image has no color information and automatically do this. That was the issue @jonnyawsom3 was referring to.

Also, in your initial post you mentioned this:
> Expected behavior
File size shouldn't be so drastically different.

This an incorrect assumption, it's never mentioned anywhere that `--disable_perceptual_optimizations` should produce similar file sizes. Just that it disables a lot of the techniques/algorithms that libjxl uses to improve image compression for human eyes. Fundamentally, the whole point of JPEG XL is to target a visual quality, not a certain file size. 

In my opinion, the real issue is exposing `--disable_perceptual_optimizations` in the CLI with no explicit warnings, because in 99.9% of cases using it is a really, really, really bad idea!

**Comment by cchang326** [comment]
> This is an incorrect assumption, it's never mentioned anywhere that --disable_perceptual_optimizations should produce similar file sizes.

I came from video compression world and switching perceptual optimizations (PSNR vs SSIM, e.g.) changes the cost function but not target bitrate or compression ratio. I guess I was incorrectly assuming the option is doing something similar.

I made a mistake when creating that gray scale image with 3 channels. I fixed that and made a gray scale image. Re-encoded it with and without "-C 0 -m 1". I'm getting similar results:

409,961 gray_C0.jxl
120,626 gray.jxl
![kodim05_gray](https://github.com/user-attachments/assets/a81f4026-1d48-4570-af42-efb89469fa1a)

**Comment by Galaxy4594** [comment]
> 409,961 gray_C0.jxl
120,626 gray.jxl

How strange, what are your encoding settings? Operating system? Libjxl version?
Here are my settings and outputs on Windows libjxl 0.11:
`cjxl input.png output.jxl` -------- 115,137 bytes
`cjxl input.png output.jxl -m 1` -- 127,057 bytes
It seems to work fine for me without `-C 0`

**Comment by cchang326** [comment]
I'm on Windows.

JPEG XL encoder v0.11.0 4df1e9e [AVX2,SSE2]
Encoding [VarDCT, d1.000, effort: 3]
Compressed to 120.6 kB (2.454 bpp).
768 x 512,  25.978 MP/s [25.98, 25.98], , 1 reps, 64 threads.

`cjxl.exe kodim05_gray.png gray.jxl -q 90 -e 3`
`cjxl.exe kodim05_gray.png gray_C0.jxl -q 90 -e 3 -C 0 -m 1`

**Comment by cchang326** [comment]
Indeed `-C 0 ` doesn't make a difference. The input image is 8bit PNG based on Windows file property. Why do you say it's RGB?

**Comment by cchang326** [comment]
Ah, looks like `-e 3` is causing the difference.

**Comment by Galaxy4594** [comment]
Yup, I can confirm. Effort 3 is the culprit, bump up modular mode to effort 4 and the file sizes are much smaller. 

`cjxl gray.png gray.jxl -m 1 -e 3` 410,015 bytes
`cjxl gray.png gray.jxl -m 1 -e 4` 134,960 bytes

Anyways, does this resolve your initial issue regarding `--disable_perceptual_optimization`?

**Comment by jonnyawsom3** [comment]
Going to close this for now due to inactivity.
